### ISSUE TITLE
Profunctor instance for (FoldM m)

### ISSUE THREAD

**Issue by treeowl** [Gabriel439/Haskell-Foldl-Library#41]
If I'm not mistaken,

```haskell
instance Monad m => Profunctor (FoldM m) where
  rmap = fmap
  lmap = premapM
```
Speaking of `premapM`, it appears that its `Monad m` context is redundant.

**Comment by Gabriel439** [comment]
Sounds good!  Do you have time to write up a pull request for this?

**Comment by treeowl** [comment]
Yes.

**Comment by Gabriel439** [comment]
Closed by #42
